### PR TITLE
Add GPRF class

### PR DESCRIPTION
**Pull request by mmamayka**
Adds General Purpose Register File implementing behavior of 
- 32 GPR registers (`x0`-`x31`)
- program counter (`pc`) register

### CODE REVIEW

File: include/besm-666/reg/rf.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#include <cassert>
+#include <cstddef>
+
+#include "besm-666/instruction.hpp"
+#include "besm-666/riscv-types.hpp"
+
+namespace besm::reg {
+
+/// General purpose register file
+class GPRF {
```
Comment (levBagryansky) [inline include/besm-666/reg/rf.hpp:13]: @mmamayka let's add macro `PC 32` to use it clearer. To not to keep in mind that 32's register is PC.

**Review by levBagryansky — COMMENTED** [review]
@mmamayka LGTM

**Review by c71n93 — DISMISSED** [review]

File: include/besm-666/reg/rf.hpp (hunk 1 above)
Comment (mmamayka) [reply include/besm-666/reg/rf.hpp:13]: Done

**Review by levBagryansky — APPROVED** [review]
@mmamayka LGBT

**Review by c71n93 — APPROVED** [review]

**Pull request merged**